src/algorithms/graphs/breadthFirstSearch.java

```diff
@@ -0,0 +1,133 @@
+package src.algorithms.graphs;
+
+import java.util.*;
+
+import src.algorithms.graphs.util.BinaryTreeNode;
+import src.algorithms.graphs.util.GraphNode;
+
+/**
+ * Implementation of BFS
+ *
+ * Breadth-First search is a graph traversal algorithm that utilizes a queue (FIFO) data structure
+ * It is useful in finding a shortest-hops solution in graphs
+ * This method is also used to obtain level order traversals of trees.
+ *
+ * In general, BFS works as such:
+ *  - Start with a queue that contains the root node
+ *  - While the queue is not empty:
+ *      - Pop a vertex from the queue
+ *      - Push all neighbours to the queue if they have not been visited yet
+ *      - Update any variables as needed
+ *
+ * Time: O(V + E), where V is the number of vertices/nodes, and E is the number of edges in the graph
+ * Explanation: Each vertex is popped in O(1) time from the stack exactly once, hence O(V)
+ *              For each edge, we must check in O(1) time if we have visited the adjacent vertex to know
+ *              whether to push it into the queue, hence O(E).
+ *              Note that if the graph is a forest, BFS will likely terminate earlier, as fewer vertices are traversed
+ *
+ * Space: O(V): We utilize a Hashset to store the vertices we have visited already. In the worst case we have a
+ *              connected graph where all vertices are traversed, and our Hashset stores all O(V) vertices.
+ *              Further, we use a queue to hold the vertices to be traversed. In the worst case, the root will have all
+ *              other vertices as neighbours, and our queue will contain O(V) vertices.
+ *
+ * ** Note: The above description assumes an adjacency list in order to consider neighbours in an efficient manner
+ *          If an adjacency matrix were used, it would cost O(V) to find neighbours for a single vertex, making our
+ *          average case time complexity O(V^2) for a connected graph
+ *
+ * The implementation demonstrates the use of BFS in finding the level-order (Root, Left, Right) traversal of a binary tree
+ * The tree is represented using a custom BinaryTreeNode class
+ *
+ */
+public class breadthFirstSearch {
+
+    // Prints level order traversal from left to right
+    public static List<Integer> levelOrder(BinaryTreeNode root) {
+        if (root == null) { return new ArrayList<>(); }
+        List<Integer> traversal = new ArrayList<>();
+        Queue<BinaryTreeNode> queue = new LinkedList<>();
+        queue.add(root);
+
+        while (!queue.isEmpty()) {
+            BinaryTreeNode curr = queue.remove();
+            traversal.add((Integer) curr.getVal());
+            if (curr.getLeft() != null) { queue.add(curr.getLeft()); }
+            if (curr.getRight() != null) { queue.add(curr.getRight()); }
+        }
+
+        return traversal;
+    }
+
+    // Finds the number of friend hops needed to go from person A to person B
```

Comment:
great explanation. But for standardisation, we use // as in-line comments to birefly explain logic or parts of the code that aren't clear. If you're explaining high-level overview eg what the method does, can use the /** */ or see how cx and kt did it!